sql/core/src/main/scala/org/apache/spark/sql/execution/AliasAwareOutputExpression.scala

```diff
@@ -65,7 +65,11 @@ trait AliasAwareOutputOrdering extends AliasAwareOutputExpression {
 
   final override def outputOrdering: Seq[SortOrder] = {
     if (hasAlias) {
-      orderingExpressions.map(normalizeExpression(_).asInstanceOf[SortOrder])
```

Comment:
shall we include `sameOrderExpressions` as children of `SortOrder`? cc @viirya @maropu

Reply:
Ah, I see. It seems we can and the refactoring looks fine to me. Do you have time to work on it, @prakharjain09, in a separate PR?

Reply:
@cloud-fan @maropu Sure. I will raise a separate PR to make sameOrderExpressions as children of sortorder - once this goes in.